rpc/lib/client/ws_client.go

```diff
@@ -76,7 +76,7 @@ type WSClient struct {
 // functions for a detailed description of how to configure ping period and
 // pong wait time. The endpoint argument must begin with a `/`.
 func NewWSClient(remoteAddr, endpoint string, options ...func(*WSClient)) *WSClient {
-	addr, dialer := makeHTTPDialer(remoteAddr)
+	_, addr, dialer := makeHTTPDialer(remoteAddr)
```

Comment:
This doesn't seem to differentiate between ws:// and wss://, which should be equivalent to the difference between http and https.

I had to specify that to get the connection working in javascript: https://github.com/confio/weave-js/blob/master/src/tendermint/rpc.js#L27-L46
maybe something similar is needed here.